base/windows/item/ClassChangeWithItemWindow.ts

```diff
@@ -0,0 +1,144 @@
+import {TextObj, Window} from "../../Window";
+import {Item} from "../../Item";
+import {GoldenSun} from "../../GoldenSun";
+import {ItemSlot, MainChar} from "../../MainChar";
+import * as _ from "lodash";
+import {choose_right_class} from "../../Classes";
+
+const BASE_WIN_WIDTH = 100;
+const BASE_WIN_HEIGHT = 92;
+const BASE_WIN_X = 0;
+const BASE_WIN_Y = 40;
+const ARROW_X = 32;
+const ARROW_Y = 49;
+
+export class ClassChangeWithItemWindow {
+    public game: Phaser.Game;
+    public data: GoldenSun;
+    public char: MainChar;
+    public window_open: boolean;
+    public x: number;
+    public y: number;
+    public base_window: Window;
+    public avatar_group: Phaser.Group;
+    public x_avatar: number;
+    public y_avatar: number;
+    public avatar: Phaser.Sprite;
+    public name_text: TextObj;
+    public lv_text: TextObj;
+    public class_text: TextObj;
+
+    public preview_class_text: TextObj;
+    public class_name_arrow: Phaser.Sprite;
+    public class_name_arrow_blink_timer: Phaser.Timer;
+
+    public item: Item;
+    public item_obj: ItemSlot;
+
+    constructor(game, data) {
+        this.game = game;
+        this.data = data;
+        this.char = null;
+        this.window_open = false;
+        this.x = BASE_WIN_X;
+        this.y = BASE_WIN_Y;
+        this.base_window = new Window(this.game, this.x, this.y, BASE_WIN_WIDTH, BASE_WIN_HEIGHT);
+        this.avatar_group = game.add.group();
+        this.avatar_group.alpha = 0;
+        this.x_avatar = this.x + 8;
+        this.y_avatar = this.y + 8;
+        this.avatar = null;
+
+        this.base_window.set_text_in_position("Lv", 48, 24);
+
+        this.name_text = this.base_window.set_text_in_position("0", 40, 8);
+        this.lv_text = this.base_window.set_text_in_position("0", 80, 24);
+        this.class_text = this.base_window.set_text_in_position("0", 8, 40);
+
+        this.preview_class_text = this.base_window.set_text_in_position("0", 8, 56);
+        this.class_name_arrow = this.base_window.create_at_group(ARROW_X, ARROW_Y, "arrow_change");
+        this.init_arrow_blinks();
+    }
+
+    private init_arrow_blinks() {
+        this.class_name_arrow_blink_timer = this.game.time.create(false);
+        this.class_name_arrow_blink_timer.loop(90, () => {
+            this.class_name_arrow.alpha = this.class_name_arrow.alpha ? 0 : 1;
+        });
+
+        this.class_name_arrow_blink_timer.start();
```

Comment:
Why do Start/Stop here and Resume down below when you can just start the timer in the "open" function?